### ISSUE TITLE
Unable to add locks to favorites and favorites crashing app

### ISSUE THREAD

**Issue by blair287** [home-assistant/android#1974]
**Home Assistant Android version:**

Latest build on here

**Android version:**

11

**Phone model:**

**Home Assistant version:**

**Last working Home Assistant release (if known):**

**Description of problem:**

Locks do not appear in favorites list on phone to set for watch.

Also removing favorites from phone seems to result in wear os app crashing on every load unless that favourite is added back.

**Traceback (if applicable, to get the logs you may refer to: https://companion.home-assistant.io/docs/troubleshooting/faqs/#android-crash-logs):**

```

```

**Screenshot of problem:**

**Additional information:**

**Comment by dshokouhi** [comment]
@blair287 first check that both phone and wear are on the latest beta version. Then please provide precise steps to reproduce including if you were unchecking the `lock` domain. As of now I cannot reproduce this but for the phone settings we need to add the `lock` domain. It was only merged yesterday so we had missed this.  It would also be great if we got the crash log to investigate. As of now I am able to remove items just fine and the wear OS app is not crashing.

**Comment by blair287** [comment]
> @blair287 first check that both phone and wear are on the latest beta version. Then please provide precise steps to reproduce including if you were unchecking the `lock` domain. As of now I cannot reproduce this but for the phone settings we need to add the `lock` domain. It was only merged yesterday so we had missed this. It would also be great if we got the crash log to investigate. As of now I am able to remove items just fine and the wear OS app is not crashing.

It was a script that was in my favorites that I had added before there was option to do it on phone. Both wear and phone on latest version.

If I add any more via phone I can remove fine but if I remove the script I already had it will cause the wear os app to crash on start unless via the phone app I add that script back.

**Comment by dshokouhi** [comment]
yea i will definitely need the logs as I cannot reproduce that, if you start over with favorites (which will need to happen in a future PR) does it work as expected?

**Comment by dshokouhi** [comment]
Also is that the only favorite you had at the time or were there others too?

**Comment by SkechyWolf** [comment]
> Also is that the only favorite you had at the time or were there others too?

I did some testing and removed all favorites through my phone then opened the app on my watch which causes the wear os app to crash with this error

When removing all favorites on phone it causes the UI on the watch to make an empty chip(I think that's what it's called) then if you open the favorite settings on the mobile app it shows a checkbox with no name set as the current favorite

Sorry if this is not of any help
```
2021-11-28 14:54:59.827 15601-15601/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: io.homeassistant.companion.android, PID: 15601
    java.lang.NullPointerException
        at io.homeassistant.companion.android.home.views.MainViewKt$MainView$3$1$2.invoke(MainView.kt:144)
        at io.homeassistant.companion.android.home.views.MainViewKt$MainView$3$1$2.invoke(MainView.kt:113)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:135)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
        at androidx.wear.compose.material.ScalingLazyListScopeImpl$items$1$1.invoke(ScalingLazyColumn.kt:436)
        at androidx.wear.compose.material.ScalingLazyListScopeImpl$items$1$1.invoke(ScalingLazyColumn.kt:435)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:116)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
        at androidx.wear.compose.material.ScalingLazyColumnKt.ScalingLazyColumnItemWrapper(ScalingLazyColumn.kt:468)
        at androidx.wear.compose.material.ScalingLazyColumnKt.access$ScalingLazyColumnItemWrapper(ScalingLazyColumn.kt:1)
        at androidx.wear.compose.material.ScalingLazyListScopeImpl$items$1.invoke(ScalingLazyColumn.kt:431)
        at androidx.wear.compose.material.ScalingLazyListScopeImpl$items$1.invoke(ScalingLazyColumn.kt:430)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:135)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
        at androidx.compose.foundation.lazy.LazyListScopeImpl$items$1$1.invoke(LazyListScopeImpl.kt:39)
        at androidx.compose.foundation.lazy.LazyListScopeImpl$items$1$1.invoke(LazyListScopeImpl.kt:39)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:107)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
        at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:228)
        at androidx.compose.runtime.saveable.SaveableStateHolderImpl.SaveableStateProvider(SaveableStateHolder.kt:84)
        at androidx.compose.foundation.lazy.layout.LazyLayoutItemContentFactory$CachedItemContent$content$1.invoke(LazyLayoutItemContentFactory.kt:102)
        at androidx.compose.foundation.lazy.layout.LazyLayoutItemContentFactory$CachedItemContent$content$1.invoke(LazyLayoutItemContentFactory.kt:93)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:107)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
        at androidx.compose.runtime.RecomposeScopeImpl.compose(RecomposeScopeImpl.kt:140)
        at androidx.compose.runtime.ComposerImpl.recomposeToGroupEnd(Composer.kt:2158)
        at androidx.compose.runtime.ComposerImpl.skipCurrentGroup(Composer.kt:2404)
        at androidx.compose.runtime.ComposerImpl$doCompose$2$5.invoke(Composer.kt:2585)
        at androidx.compose.runtime.ComposerImpl$doCompose$2$5.invoke(Composer.kt:2571)
        at androidx.compose.runtime.SnapshotStateKt__DerivedStateKt.observeDerivedStateRecalculations(DerivedState.kt:234)
        at androidx.compose.runtime.SnapshotStateKt.observeDerivedStateRecalculations(Unknown Source:1)
        at androidx.compose.runtime.ComposerImpl.doCompose(Composer.kt:2571)
        at androidx.compose.runtime.ComposerImpl.recompose$runtime_release(Composer.kt:2547)
        at androidx.compose.runtime.CompositionImpl.recompose(Composition.kt:620)
        at androidx.compose.runtime.Recomposer.performRecompose(Recomposer.kt:786)
        at androidx.compose.runtime.Recomposer.access$performRecompose(Recomposer.kt:105)
        at androidx.compose.runtime.Recomposer$runRecomposeAndApplyChanges$2$2.invoke(Recomposer.kt:456)
        at androidx.compose.runtime.Recomposer$runRecomposeAndApplyChanges$2$2.invoke(Recomposer.kt:425)
        at androidx.compose.ui.platform.AndroidUiFrameClock$withFrameNanos$2$callback$1.doFrame(AndroidUiFrameClock.android.kt:34)
        at androidx.compose.ui.platform.AndroidUiDispatcher.performFrameDispatch(AndroidUiDispatcher.android.kt:109)
2021-11-28 14:54:59.827 15601-15601/? E/AndroidRuntime:     at androidx.compose.ui.platform.AndroidUiDispatcher.access$performFrameDispatch(AndroidUiDispatcher.android.kt:41)
        at androidx.compose.ui.platform.AndroidUiDispatcher$dispatchCallback$1.doFrame(AndroidUiDispatcher.android.kt:69)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:996)
        at android.view.Choreographer.doCallbacks(Choreographer.java:796)
        at android.view.Choreographer.doFrame(Choreographer.java:727)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:983)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:246)
        at android.app.ActivityThread.main(ActivityThread.java:7690)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:593)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:995)
```

**Comment by dshokouhi** [comment]
thanks @SkechyWolf this is helpful :) i'll add a fix to my current PR since favorites will need to be reset anyways

**Comment by blair287** [comment]
> thanks @SkechyWolf this is helpful :) i'll add a fix to my current PR since favorites will need to be reset anyways
> 
> Edit: Fixed in [0ee6f34](https://github.com/home-assistant/android/commit/0ee6f34df1ccc1f9d017ddaabed986c8839662d4)
> 
> Good catch :)

Yes the above was the same issue sorry I didn't get logs back on shift now so didn't have much time.

Awesome work!
